[DOCS] Updates the "Interactive Mode" guide for creating Expectations

Rachel-Reverie opened the pull request:
### Changes proposed in this pull request:
- update "create Expectations with instant feedback" CLI guide to a Jupyter Notebook based guide that works with fluent-config Datasources

### Definition of Done
- [X] I have made corresponding changes to the documentation
- [X] I have run any local integration tests and made sure that nothing is broken.

Rachel-Reverie committed:
doc-517
prerequisites and intial outline of steps for guide

Rachel-Reverie committed:
- additional draft work

Rachel-Reverie committed:
- Rewrites the guide "how to create and edit expectations in python with instant feedback from a sample batch of data" to a fluent-config Datasource friendly method

ghost commented:
<details open>
<summary><b>👇 <a href="https://app.codesee.io/r/reviews?pr=7624&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations">Click on the image</a> for a new way to code review</b>
</summary>

####

[![Review these changes using an interactive CodeSee Map](https://s3.us-east-2.amazonaws.com/maps.codesee.io/images/github/great-expectations/great_expectations/7624/a0d51e11/5535c906d855fe96089dd60aae14578b221735fb.svg)](https://app.codesee.io/r/reviews?pr=7624&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations)

#### Legend
<img src="https://app.codesee.io/Legend_v3.png" alt="CodeSee Map legend" width="220" height="156">
</details>

Rachel-Reverie committed:
- renames file for guide to match original
- removes titles from code blocks (they are apparently not compatible with named snippets).

Rachel-Reverie committed:
- adds link for creating block-config Datasource compatible Batch Requests

kwcanuck commented on docs/docusaurus/docs/guides/expectations/components_how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data/_preface.mdx:
```diff
@@ -10,3 +10,39 @@ This guide will take you through the process of creating <TechnicalTag tag="expe
+
+## Steps
+
+### 1. Import Great Expectations and get a Data Context
```
Why are you using empty headings instead of a simple procedure? This information would be better presented as a procedure. All of the headings make it visually cluttered.

Rachel-Reverie replied:
The use of headings was to build out the right hand navigation, which is automatically built by Docusaurus based on headers.  I think we should revisit this, update our doc standards, and then standardize the changes as it's own work - revising our current style guidance would be out of scope for this PR.

kwcanuck commented on docs/docusaurus/docs/guides/expectations/components_how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data/_preface.mdx:
```diff
@@ -10,3 +10,39 @@ This guide will take you through the process of creating <TechnicalTag tag="expe
 
 </Prerequisites>
+
+## Steps
```
```suggestion
## Steps
```
I really dislike _Steps_.  It's a personal bias. What will users achieve after completing this procedure? Create a meaningful heading.

Rachel-Reverie replied:
Let's set up an information architecture review meeting and revise our standards for formatting how-to guides -- making a change to the current standards is out of scope for this PR.

kwcanuck commented on docs/docusaurus/docs/guides/expectations/components_how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data/_preface.mdx:
```diff
@@ -10,3 +10,39 @@ This guide will take you through the process of creating <TechnicalTag tag="expe
+
+:::info Working outside of a Jupyter Notebook
+
+In a Jupyter notebook, the results of a given cell will automatically be displayed when the cell is run.  If you are working in a different interpreter, you may need to explicitly print these results, instead:
```
```suggestion
In a Jupyter Notebook, the results for a given cell are automatically displayed when you run the cell.  If you're using a different interpreter, you might need to add the following method to explicitly print these results:
```

kwcanuck commented on docs/docusaurus/docs/guides/expectations/components_how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data/_preface.mdx:
```diff
@@ -10,3 +10,39 @@ This guide will take you through the process of creating <TechnicalTag tag="expe
+
+:::info Working outside of a Jupyter Notebook
+
+Again, if you are not working in a Jupyter Notebook you may need to explicitly print your results:
```
```suggestion
If you're using a different interpreter, you might need to add the following method to explicitly print these results:
```

kwcanuck commented on docs/docusaurus/docs/guides/expectations/components_how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data/_preface.mdx:
```diff
@@ -10,3 +10,39 @@ This guide will take you through the process of creating <TechnicalTag tag="expe
+### 5. Repeat step 4 to edit the Expectation's values or add additional Expectations
+
+
+### 6. Save your Expectations for future use
```
```suggestion
### 6. Save your Expectations
```

kwcanuck commented on docs/docusaurus/docs/guides/expectations/components_how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data/_preface.mdx:
```diff
@@ -10,3 +10,39 @@ This guide will take you through the process of creating <TechnicalTag tag="expe
+
+:::
+
+### 5. Repeat step 4 to edit the Expectation's values or add additional Expectations
```
```suggestion
### 5. Repeat step 4 to edit Expectation values or add additional Expectations
```

kwcanuck commented on docs/docusaurus/docs/guides/expectations/how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data.md:
```diff
@@ -1,48 +1,158 @@
-title: How to create and edit Expectations with instant feedback from a sample Batch of data
+title: How to create Expectations interactively in Python
+tag: [how-to, getting started]
+description: A how-to guide describing how to create Expectations in a Python interpreter or script while interactively receiving feedback by validating them against a Batch of data.
```
```suggestion
description: Create Expectations with a Python interpreter or a script and then use interactive feedback to validate them with batch data.
```

kwcanuck commented on docs/docusaurus/docs/guides/expectations/how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data.md:
```diff
@@ -1,48 +1,158 @@
+import ConnectingToDataFluently from '/docs/components/connect_to_data/link_lists/_connecting_to_data_fluently.md'
 
-<Preface />
+In order to Validate data, we must first define a set of Expectations for that data to be validated against.  In this guide, we will walk you through processes of interactively creating and editing Expectations while simultaneously validating them against a Batch of data. By validating our Expectations as we define them we can quickly see if the Expectations we are setting are suitable for our da ...
```
```suggestion
To validate data, you need to first define the Expectations that you'll use for your data validation.  In this guide, you'll learn how to interactively create and edit Expectations and then validate them with batch data. Validating your Expectations as you define them allows you to quickly determine if the Expectations are suitable for your data, and identify where changes might be necessary.
```

kwcanuck commented on docs/docusaurus/docs/guides/expectations/how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data.md:
```diff
@@ -1,48 +1,158 @@
-<Preface />
+In order to Validate data, we must first define a set of Expectations for that data to be validated against.  In this guide, we will walk you through processes of interactively creating and editing Expectations while simultaneously validating them against a Batch of data. By validating our Expectations as we define them we can quickly see if the Expectations we are setting are suitable for our da ...
+
+:::tip Does this process edit my data?
```
I would argue that this information shouldn't be in a tip admonition. It's not a tip. It's simply informative. Punctuation in headings should be used sparingly, if at all.

kwcanuck commented on docs/docusaurus/docs/guides/expectations/how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data.md:
```diff
@@ -1,48 +1,158 @@
+In order to Validate data, we must first define a set of Expectations for that data to be validated against.  In this guide, we will walk you through processes of interactively creating and editing Expectations while simultaneously validating them against a Batch of data. By validating our Expectations as we define them we can quickly see if the Expectations we are setting are suitable for our da ...
+
+:::tip Does this process edit my data?
+Not at all.  We call this process the "Interactive method" of creating and editing Expectations because validating the Expectations as we work requires interacting with a Batch of data.  This data is introspected to see if it fits the defined Expectation, but it is not edited: only examined.
```
```suggestion
No.  The interactive method used to create and edit Expectations does not edit or alter the batch data.
```
If you don't accept the edit, I recommend replacing _introspected_ with a simpler term.

kwcanuck commented on docs/docusaurus/docs/guides/expectations/how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data.md:
```diff
@@ -1,48 +1,158 @@
+<Prerequisites>
+
+- Great Expectations installed in a Python environment
+- Initialized a Filesystem Data Context to save your Expectations in
```
```suggestion
- A Filesystem Data Context for your Expectations
```
If you don't accept the edit, I recommend finding a simpler term than _Initialized_

kwcanuck commented on docs/docusaurus/docs/guides/expectations/how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data.md:
```diff
@@ -1,48 +1,158 @@
+
+- Great Expectations installed in a Python environment
+- Initialized a Filesystem Data Context to save your Expectations in
+- Created a Datasource from which to request a Batch of data for introspection
```
```suggestion
- A Datasource that you can use to request and review batch data
```
If you don't accept the edit, I recommend replacing _introspection_ with a simpler term.

kwcanuck commented on docs/docusaurus/docs/guides/expectations/how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data.md:
```diff
@@ -1,48 +1,158 @@
+- Great Expectations installed in a Python environment
+- Initialized a Filesystem Data Context to save your Expectations in
+- Created a Datasource from which to request a Batch of data for introspection
+- A passion for data quality
```
```suggestion
- A passion for data quality
```
I would remove this

kwcanuck commented on docs/docusaurus/docs/guides/expectations/how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data.md:
```diff
@@ -1,48 +1,158 @@
+<details>
+<summary>
+
+### If you still need to set up Great Expectations...
```
```suggestion
### If you haven't set up Great Expectations
```

kwcanuck commented on docs/docusaurus/docs/guides/expectations/how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data.md:
```diff
@@ -1,48 +1,158 @@
+<details>
+<summary>
+
+### If you still need to initialize your Data Context...
```
```suggestion
### If you haven't initialized your Data Context
```

kwcanuck commented on docs/docusaurus/docs/guides/expectations/how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data.md:
```diff
@@ -1,48 +1,158 @@
+
+</summary>
+
+Please see the appropriate guide from the following:
```
```suggestion
Select one of the following guides:
```

kwcanuck commented on docs/docusaurus/docs/guides/expectations/how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data.md:
```diff
@@ -1,48 +1,158 @@
+
+</summary>
+
+Please reference the appropriate guide from the following:
```
```suggestion
Select one of the following guides:
```
Is there an inappropriate guide? :)

![image](https://user-images.githubusercontent.com/104205257/231878510-57d9f538-739c-4867-97e7-2d1ef6b554b2.png)

Rachel-Reverie replied:
The inappropriate guides would be the guides that correspond to environments or source data systems that don't match the one they're using. :P

kwcanuck commented on docs/docusaurus/docs/guides/expectations/how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data.md:
```diff
@@ -1,48 +1,158 @@
-<UseTheCliToBeginTheInteractiveProcessOfCreatingExpectations />
+### 1. Import the Great Expectations module and instantiate a Data Context
+
+For this guide we will be working with Python code in a Jupyter Notebook. Jupyter is included with GX and provides a very convenient interface that lets us easily edit code and immediately see the result of our changes.
```
```suggestion
To complete the procedures in this guide, you'll use Python code in a Jupyter Notebook. Jupyter is included with GX and it lets you edit code and immediately see the results of your changes.
```
What are users doing here? Tell them what they'll accomplish - _You'll use Python code in a Jupyter Notebook to..._

![image](https://user-images.githubusercontent.com/104205257/231879477-9680f7b3-d0b8-488b-abf0-38c372296fec.png)

kwcanuck commented on docs/docusaurus/docs/guides/expectations/how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data.md:
```diff
@@ -1,48 +1,158 @@
+
+For this guide we will be working with Python code in a Jupyter Notebook. Jupyter is included with GX and provides a very convenient interface that lets us easily edit code and immediately see the result of our changes.
+
+The code to import Great Expectations and instantiate a Data Context is:
```
```suggestion
Run the following code to import Great Expectations and instantiate a Data Context:
```

kwcanuck commented on docs/docusaurus/docs/guides/expectations/how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data.md:
```diff
@@ -1,48 +1,158 @@
+
+:::info Data Contexts and persisting data
+
+If you are using an Ephemeral Data Context, your configurations will not persist beyond the current Python session.  If you are using a Filesystem or Cloud Data Context, they will.  The `get_context()` method will return the first of a Cloud or Filesystem Data Context if it can find one, but if it can't then it will provide you with an Ephemeral Data Context.  For more information on the `get_con ...
```
```suggestion
If you're using an Ephemeral Data Context, your configurations won't persist beyond the current Python session.  However, if you're using a Filesystem or Cloud Data Context, they do persist.  The `get_context()` method returns the first Cloud or Filesystem Data Context it can find. If a Cloud or Filesystem Data Context can't be located, it provides an Ephemeral Data Context. For more information about the `get_context()` method, see [How to quickly instantiate a Data Context](/docs/guides/setup/configuring_data_contexts/instantiating_data_contexts/how_to_quickly_instantiate_a_data_context).
```

kwcanuck commented on docs/docusaurus/docs/guides/expectations/how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data.md:
```diff
@@ -1,48 +1,158 @@
+
+### 2. Use an existing Data Asset to create a Batch Request
+
+From the Data Context we initialized we can access our previously created Datasources and Data Assets.  We will retrieve a previously configured Data Asset, and then create a Batch Request indicating the Batch of data that we intend to Validate our Expectations against:
```
```suggestion
Add the following method to retrieve a previously configured Data Asset from the Data Context you initialized and create a Batch Request to identify the batch data that you'll use to validate your Expectations:
```

kwcanuck commented on docs/docusaurus/docs/guides/expectations/how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data.md:
````diff
@@ -1,48 +1,158 @@
+```python name="tests/integration/docusaurus/validation/validator/how_to_create_and_edit_expectations_with_instant_feedback_fluent.py get_data_asset_and_build_batch_request"
+```
+
+:::info Limiting the Batches returned by a Batch Request
````
```suggestion
:::info Limit the Batches returned by a Batch Request
```

kwcanuck commented on docs/docusaurus/docs/guides/expectations/how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data.md:
```diff
@@ -1,48 +1,158 @@
+
+:::info Limiting the Batches returned by a Batch Request
+
+A Batch Request allows you to limit the Batches returned from a Data Asset by providing a dictionary as the `options` parameter.  If nothing is provided as the `options` parameter, your Batch Request will include all the Batches configured in the Data Asset.  To learn more, please see our guide on [how to request data from a Data Asset](/docs/guides/connecting_to_your_data/fluent/batch_requests/h ...
```
```suggestion
You can provide a dictionary as the `options` parameter to limit the Batches returned from a Data Asset.  If you leave the `options` parameter empty, your Batch Request includes all of the Batches configured in the Data Asset.  For more information about Batch Requests, see [How to request data from a Data Asset](/docs/guides/connecting_to_your_data/fluent/batch_requests/how_to_request_data_from_a_data_asset).
```

kwcanuck commented on docs/docusaurus/docs/guides/expectations/how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data.md:
```diff
@@ -1,48 +1,158 @@
+
+### 3. Create a Validator
+
+We will use a Validator to interactively create our Expectations.  To do this, a Validator needs two parameters: one will indicate the data to run Expectations against, and the other will provide a name for the combined list of Expectations should we decide to save them for future use.
```
```suggestion
To use a Validator to interactively create your Expectations, the Validator needs two parameters. One parameter indicates the batch data that is used to validate the Expectations. The other parameter provides a name for the combined list of Expectations that you can save for future use.
```

kwcanuck commented on docs/docusaurus/docs/guides/expectations/how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data.md:
```diff
@@ -1,48 +1,158 @@
+
+:::info Working outside a Jupyter Notebook
+
+In a Jupyter Notebook you will automatically see the results of the above code in a new cell when the code is run.  However, if you are working with a different Python interpreter you may need to explicitly print your results:
```
```suggestion
If you're using a Jupyter Notebook you'll automatically see the results of the code you run in a new cell when you run the code. If you're using a different interpreter, you might need to add the following method to explicitly print these results:
```

Rachel-Reverie committed:
Update docs/docusaurus/docs/guides/expectations/components_how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data/_preface.mdx

Co-authored-by: Rob Gray <104205257+kwcanuck@users.noreply.github.com>

kwcanuck commented on docs/docusaurus/docs/guides/expectations/how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data.md:
```diff
@@ -1,48 +1,158 @@
+
+### 4. Use the Validator to create and run an Expectation
+
+The Validator provides access to all of our available Expectations as methods.  By running an `expect_*()` method from the Validator, two things will happen: The Validator will add the specified Expectation to an Expectation Suite (or edit an existing Expectation in the Expectation Suite, if applicable) in its configuration, and the specified Expectation will be run against the data we provided t ...
```
```suggestion
The Validator provides access to all of the available Expectations as methods.  When you run an `expect_*()` method from the Validator, the Validator adds the specified Expectation to an Expectation Suite (or edits an existing Expectation in the Expectation Suite, if applicable) in its configuration, and then the specified Expectation is run against the data you provided to the Validator with your Batch Request.
```

Rachel-Reverie committed:
Update docs/docusaurus/docs/guides/expectations/components_how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data/_preface.mdx

Co-authored-by: Rob Gray <104205257+kwcanuck@users.noreply.github.com>

Rachel-Reverie committed:
Update docs/docusaurus/docs/guides/expectations/how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data.md

Co-authored-by: Rob Gray <104205257+kwcanuck@users.noreply.github.com>

kwcanuck commented on docs/docusaurus/docs/guides/expectations/how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data.md:
````diff
@@ -1,48 +1,158 @@
+```python name="tests/integration/docusaurus/validation/validator/how_to_create_and_edit_expectations_with_instant_feedback_fluent.py interactive_validation"
+```
+
+Since we are working in a Jupyter Notebook, the results of the Validation will be printed after we run an `expect_*()` method.  We can examine those results to determine if the Expectation needs to be edited (by running it again and providing additional optional parameters, or different parameter values).
````
```suggestion
If you're using a Jupyter Notebook, the results of the Validation are printed after you run an `expect_*()` method.  You can examine the results to determine if the Expectation needs to be edited. To edit an Expectation, you run it again with additional optional parameters, or different parameter values.
```

kwcanuck commented on docs/docusaurus/docs/guides/expectations/how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data.md:
```diff
@@ -1,48 +1,158 @@
+Since we are working in a Jupyter Notebook, the results of the Validation will be printed after we run an `expect_*()` method.  We can examine those results to determine if the Expectation needs to be edited (by running it again and providing additional optional parameters, or different parameter values).
+
+:::info Working outside a Jupyter Notebook
+As before, if you are not working in a Jupyter Notebook you may need to explicitly print your results:
```
```suggestion
If you're using a different interpreter, you might need to add the following method to explicitly print these results:
```

kwcanuck commented on docs/docusaurus/docs/guides/expectations/how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data.md:
```diff
@@ -1,48 +1,158 @@
+
+:::
+
+### 5. (Optional) Repeat step 4 to create additional Expectations or edit existing ones
```
```suggestion
### 5. (Optional) Repeat step 4 to create additional Expectations or edit Expectations
```

Rachel-Reverie committed:
Update docs/docusaurus/docs/guides/expectations/components_how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data/_preface.mdx

Co-authored-by: Rob Gray <104205257+kwcanuck@users.noreply.github.com>

Rachel-Reverie committed:
Update docs/docusaurus/docs/guides/expectations/components_how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data/_preface.mdx

Co-authored-by: Rob Gray <104205257+kwcanuck@users.noreply.github.com>

kwcanuck commented on docs/docusaurus/docs/guides/expectations/how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data.md:
```diff
@@ -1,48 +1,158 @@
+
+### 5. (Optional) Repeat step 4 to create additional Expectations or edit existing ones
+
+After reviewing the Validation Results that were returned when we created an Expectation, we can choose to edit that Expectation by running the `validator.expect_*()` method with different parameters than previously.  We can also have the Validator run an entirely different `expect_*()` method and create additional Expectations.  All the Expectations that we create will be stored in a list in the ...
```
```suggestion
After you've reviewed the Validation Results that were returned when you created an Expectation, you can run the `validator.expect_*()` method with different parameters to edit the Expectation.  You can also have the Validator run an entirely different `expect_*()` method to create additional Expectations.  All the Expectations that you create are stored in a list in the Validator configuration file.
```

kwcanuck commented on docs/docusaurus/docs/guides/expectations/how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data.md:
```diff
@@ -1,48 +1,158 @@
+
+:::tip What if I want to use the same Expectation more than once?
+
+GX takes into account certain parameters when determining if an Expectation is being added to the list or if an existing Expectation should be edited.  For example, if you are using an expectation such as `expect_column_*()` you would edit an existing Expectation by providing the same `column` parameter when repeating the previous steps, and different values for any other parameters.  If you used ...
```
```suggestion
GX takes into account certain parameters when determining if an Expectation is being added to the list or if an existing Expectation should be edited.  For example, if you are using an expectation such as `expect_column_*()` you would edit an existing Expectation by providing the same `column` parameter when repeating the previous steps, and different values for any other parameters.  If you used a different value for the `column` parameter, you will create an additional instance of the Expectation for the new `column` value, rather than overwrite the Expectation you defined with the first `column` value.
```
 _when repeating the previous steps_ - What specific steps?

I think this needs some simplification and clarification

kwcanuck commented on docs/docusaurus/docs/guides/expectations/how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data.md:
```diff
@@ -1,48 +1,158 @@
 
-### 4. Continue the workflow within a Jupyter Notebook
-<ContinueTheWorkflowWithinAJupyterNotebook />
+The Expectations we have created so far are saved in an Expectation Suite on the Validator object.  Since Validators do not persist outside the current Python session these Expectations will not be kept if we stop now.  This can be ideal if we are using a Validator to do some quick data discovery, but in most cases we will want our newly created Expectation Suite to be available in the future.
```
```suggestion
The Expectations you created are saved in an Expectation Suite on the Validator object.  Validators do not persist outside the current Python session and for this reason the Expectations are not kept unless **<you do what>**?  This can be ideal if you are using a Validator for quick data validation, but in most cases you'll want to reuse your newly created Expectation Suite.
```

kwcanuck commented on docs/docusaurus/docs/guides/expectations/how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data.md:
```diff
@@ -1,48 +1,158 @@
+The Expectations we have created so far are saved in an Expectation Suite on the Validator object.  Since Validators do not persist outside the current Python session these Expectations will not be kept if we stop now.  This can be ideal if we are using a Validator to do some quick data discovery, but in most cases we will want our newly created Expectation Suite to be available in the future.
 
-<Congrats />
+To keep our Expectations for future use, we will save them to our Data Context.  A Filesystem or Cloud Data Context will persist outside the current Python session, so saving the Expectation Suite in our Data Context's Expectations Store will ensure we can access it in the future:
```
```suggestion
To keep your Expectations for future use, you save them to your Data Context.  A Filesystem or Cloud Data Context persists outside the current Python session, so saving the Expectation Suite in your Data Context's Expectations Store ensures you can access it in the future:
```

kwcanuck commented on docs/docusaurus/docs/guides/expectations/how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data.md:
```diff
@@ -1,48 +1,158 @@
 
-### 2. (Optional) Profile your data to generate Expectations, then edit them in Interactive Mode.
-<OptionalProfileYourDataToGenerateExpectationsThenEditThemInInteractiveMode />
+Ephemeral Data Contexts do not persist beyond the current Python session.  If you are working with an Ephemeral Data Context, you will need to convert it to a Filesystem Data Context using the Data Context's `convert_to_file_context()` method.  Otherwise, your saved configurations will not be available in future Python sessions as the Data Context itself will no longer be available.
```
```suggestion
Ephemeral Data Contexts don't persist beyond the current Python session.  If you're working with an Ephemeral Data Context, you'll need to convert it to a Filesystem Data Context using the Data Context's `convert_to_file_context()` method.  Otherwise, your saved configurations won't be available in future Python sessions as the Data Context itself is no longer available.
```

kwcanuck commented on docs/docusaurus/docs/guides/expectations/how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data.md:
```diff
@@ -1,48 +1,158 @@
 
-### 2. Use the built-in help to review the CLI's `suite new` optional flags
-<UseTheBuiltInHelpToReviewTheCliSSuiteNewOptionalFlags />
+Now that you have created and saved an Expectation Suite, you are ready to begin [Validating your data](/docs/guides/validation/validate_data_overview).
```
```suggestion
Now that you have created and saved an Expectation Suite, you can [Validate your data](/docs/guides/validation/validate_data_overview).
```

Rachel-Reverie committed:
- style updates per guidance

Rachel-Reverie committed:
Update docs/docusaurus/docs/guides/expectations/how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data.md

Co-authored-by: Rob Gray <104205257+kwcanuck@users.noreply.github.com>

Rachel-Reverie committed:
- style updates per guidance

Rachel-Reverie committed:
- final pass edits

Rachel-Reverie committed:
Merge branch 'develop' into d/doc-517/update_create_expectations_with_instant_feedback_cli_guide_to_notebook_based_guide
